### PR TITLE
🌱 Prepare main branch for v1.9 development

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Prepares the main branch for v1.9 development.

I took it over this time to simplify the instructions on how to actually do this correctly. So it's easier the next time for other folks to take this task.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

This is part of the new release tracking issue for v1.9 (which doesn't exist yet and will be created by the new release lead soon)

### CODE REVIEW

**Comment by sbueringer** [comment]
cc @Sunnatillo (just fyi, as you are the new release lead for v1.9 and it's one of the first tasks on the new release tracking issue)

File: test/e2e/config/docker.yaml [hunk 1]
```diff
@@ -154,7 +163,16 @@ providers:
       new: --metrics-addr=:8080
     files:
     - sourcePath: "../data/shared/v1.7/metadata.yaml"
-  - name: v1.8.99 # next; use manifest from source files
+  - name: "{go://sigs.k8s.io/cluster-api@v1.8}" # latest published release in the v1beta1 series; this is used for v1beta1 --> main clusterctl upgrades test only.
```
Comment (chrischdi) [inline test/e2e/config/docker.yaml:166]: Comment in line 157 is wrong (also for other providers)

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -97,8 +97,10 @@ is used for e.g. local development and e2e tests. We also modify tests so that t
...
 
 1. Setup E2E tests for the new release:
-   1. Goal is that we have clusterctl upgrade tests for the latest stable versions of each contract / for each supported branch. For `v1.6` this means:
-      * v1beta1: `v1.0`, `v1.4`, `v1.5` (will change with each new release)
+   1. Goal is that we have clusterctl upgrade tests for all relevant upgrade cases
+       1. Modify the test specs in `test/e2e/clusterctl_upgrade_test.go`. Please note the comments above each test case (look for `This test should be changed during "prepare main branch"`)
+          Please note that both `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` should be the highest management cluster version supported by the respective Cluster API version.
+       2. Please ping maintainers after these changes are made for a first round of feedback before continuing with the steps below.
    2. Update providers in `docker.yaml`:
        1. Add a new `v1.6.0` entry.
```
Comment (chrischdi) [inline docs/release/release-tasks.md:105]:
```suggestion
       1. Add a new `v1.6` entry.
```

File: test/e2e/data/infrastructure-docker/v1.8/bases/mp.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (chrischdi) [inline test/e2e/data/infrastructure-docker/v1.8/bases/mp.yaml:1]: looks unused?

**Review by chrischdi — COMMENTED** [review]
small nits, lgtm otherwise

File: test/e2e/config/docker.yaml (hunk 1 above)
Comment (sbueringer) [reply test/e2e/config/docker.yaml:166]: I'll drop all these comemnts. Waste of time to keep them up-to-date

**Comment by sbueringer** [comment]
One failed test, but it was an unrelated flake. Pushed a fix commit, PTAL :)

**Pull request merged**